standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/HiveJDBCConnectorProvider.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.metastore.dataconnector.jdbc;
+
+import org.apache.hadoop.hive.metastore.ColumnType;
+import org.apache.hadoop.hive.metastore.api.DataConnector;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class HiveJDBCConnectorProvider extends AbstractJDBCConnectorProvider {
+  private static Logger LOG = LoggerFactory.getLogger(HiveJDBCConnectorProvider.class);
+
+  private static final String DRIVER_CLASS = "org.apache.hive.jdbc.HiveDriver".intern();
+
+  public HiveJDBCConnectorProvider(String dbName, DataConnector dataConn) {
+    super(dbName, dataConn, DRIVER_CLASS);
+  }
+
+  /**
+   * Returns a list of all table names from the remote database.
+   * @return List A collection of all the table names, null if there are no tables.
+   * @throws MetaException To indicate any failures with executing this API
+   */
+  @Override protected ResultSet fetchTableNames() throws MetaException {
+    ResultSet rs = null;
```

Comment:
this is dupe in 3 classes:
here, DerbySQLConnectorProvider.java, MySQLConnectorProvider.java

may be we should consider a refactor?

Reply:
we can use 2 new variables for catalog/schema and schema/db and use those or introduce 2 addional methods that subclasses have to implement getSchema(), getDatabase().
How did you want me to refactor?

IMHO, this is the reason we use abstraction so subclasses can provide their implementation.

Reply:
I would have gone with adding the methods, but no hard feelings if you want to go ahead with another approach

Reply:
fixed